app/controllers/api/irs_attempts_api_controller.rb

```diff
@@ -62,6 +58,52 @@ def create
 
     private
 
+    def buffer_range(current_buffer_index:, buffer_size:, file_size:)
+      buffer_end = [current_buffer_index + buffer_size, file_size].min
+      "bytes=#{current_buffer_index}-#{buffer_end}"
+    end
+
+    def serve_s3_response(log_file_record:)
+      if IdentityConfig.store.irs_attempt_api_aws_s3_stream_enabled
+        response = s3_helper.s3_client.head_object(
+          bucket: s3_helper.attempts_bucket_name,
+          key: log_file_record.filename,
+        )
+
+        requested_data_size = response.content_length
```

Comment:
Not for this PR, but it occurs to me that we're the ones generating this file before putting it in S3, so we could just store `content_length` on the IrsAttemptApiLogFile record and avoid the `head_object` call.

Reply:
This is a really good idea!